src/Forms/RequiredFields.php

```diff
@@ -104,16 +104,29 @@ public function php($data)
                 $fieldName = $fieldName->getName();
             } else {
                 $formField = $fields->dataFieldByName($fieldName);
+                if (is_null($formField)) {
+                    $formField = $fields->dataFieldByName($fieldName . 'ID');
+                }
             }
 
             // submitted data for file upload fields come back as an array
-            $value = isset($data[$fieldName]) ? $data[$fieldName] : null;
+            $value = null;
+            if (isset($data[$fieldName])) {
+                $value = $data[$fieldName];
+            } elseif (isset($data[$fieldName . 'ID'])) {
+                $value = $data[$fieldName . 'ID'];
+            }
 
             if (is_array($value)) {
                 if ($formField instanceof FileField && isset($value['error']) && $value['error']) {
                     $error = true;
                 } else {
-                    $error = (count($value ?? [])) ? false : true;
+                    if (is_a($formField, HasOneRelationFieldInterface::class) && isset($value['value'])) {
+                        $stringValue = (string) $value['value'];
+                        $error = in_array($stringValue, ['0', '']);
+                    } else {
+                        $error = (count($value ?? [])) ? false : true;
+                    }
```

Comment:
This doesn't seem related based on the commit message. It also doesn't seem necessary? Can you please provide steps to test this? Validation seems to work without it.

Reply:
Have updated commit message. I've removed the 'ID' suffix bits from this PR.